ql/src/java/org/apache/hadoop/hive/ql/DriverFactory.java

```diff
@@ -40,7 +40,7 @@ private DriverFactory() {
   }
 
   public static IDriver newDriver(HiveConf conf) {
-    return newDriver(getNewQueryState(conf), null);
+    return newDriver(getNewQueryState(conf), QueryInfo.getFromConf(conf));
```

Comment:
I am not sure we should change this

Reply:
checked, we need this
we use this method extensively, mostly in tests, without this part the attached qtest is ruined, because query_history.start_time becomes NULL
<img width="697" alt="Screenshot 2025-01-21 at 14 23 24" src="https://github.com/user-attachments/assets/a2b359bc-1d25-47ee-a2f6-3ae80cfce2ed" />

I think any codepath that worked with a null QueryInfo should work properly with a non-null one